test/remote/gateways/remote_braintree_blue_test.rb

```diff
@@ -274,6 +274,30 @@ def test_successful_credit_card_verification
     assert_equal 'P', response.avs_result['code']
   end
 
+  def test_graceful_failure_credit_card_verification_without_zip_code
```

Comment:
This test includes the `billing_address` hash, but I think the error scenario is what occurs when there is no `billing_address` present in `options`

That's why the NoMethodError occurs when we try to access `[:zip]`

We need a test scenario that shows the transaction fails gracefully when `billing_address` returns `nil`

Reply:
ahhh gotcha, ill get it resorted!